plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -107,7 +109,7 @@ protected Control createDialogArea(final Composite parent) {
           @Override
           public int getMessageType(ValidationStatusProvider statusProvider) {
             int type = super.getMessageType(statusProvider);
-            setValid(type != IMessageProvider.ERROR);
+            setValid(type != IMessageProvider.ERROR && content.hasSelection());
```

Comment:
I agree.  The `hasSelection()` test should be moved into the `StandardDeployPreferencesPanel`.  I think the easiest is to add a _ValidationStatusProvider_ just as `setupAccountEmailDataBinding()` does.